instrumentation/aws_lambda/lib/opentelemetry/instrumentation/aws_lambda/handler.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module AwsLambda
+      # Handler class that creates a span around the _HANDLER
+      class Handler
+        attr_reader :handler_method, :handler_class
+
+        def initialize
+          @flush_timeout = ENV.fetch('OTEL_INSTRUMENTATION_AWS_LAMBDA_FLUSH_TIMEOUT', '30000').to_i
+        end
+
+        # Extract context from request headers
+        def call_wrapped(event:, context:)
+          parent_context   = extract_parent_context(event)
+          span_attributes  = event['version'] == '2.0' ? v2_proxy_attributes(event) : v1_proxy_attributes(event)
+          original_handler = resolve_original_handler
+          response         = call_original_handler(event: event, context: context)
+
+          span_attributes[OpenTelemetry::SemanticConventions::Trace::HTTP_STATUS_CODE] = response['statusCode'] if response.instance_of?(Hash) && response['statusCode']
+          span_attributes[OpenTelemetry::SemanticConventions::Resource::FAAS_ID]       = context.invoked_function_arn
+          span_attributes[OpenTelemetry::SemanticConventions::Trace::FAAS_EXECUTION]   = context.aws_request_id
+
+          OpenTelemetry::Context.with_current(parent_context) do
+            span = tracer.start_span(
+              original_handler,
+              attributes: span_attributes,
+              kind: :server # Span kind MUST be `:server` for a HTTP server span
+            )
+          rescue Exception => e # rubocop:disable Lint/RescueException
```

Comment:
Should we rescue in function level rather than in context level?

Reply:
Yes, I think that's a good call.